File: test/main/test_privacy_policy.py
from flaskapp import models
from test.main.base_classes import BaseUser
from test.main.utils import test_post_request


class PrivacyPolicyTestCase(BaseUser):
    def test_privacy_policy(self):
        response = self.client.get("/privacy-policy")
        self.assertIn(b"Your privacy is important to us.", response.data)
        self.assertIn(
            b"We only ask for personal information when we truly need it", response.data
        )
        self.assertIn(
            b"We only retain collected information for as long as", response.data
        )
        self.assertIn(
            b"necessary to provide you with your requested service.", response.data
        )
        self.assertIn(b"This policy is effective as of 9 April 2020.", response.data)
        self.assertIn(
            b"Our website may link to external sites that are not", response.data
        )
        self.assertIn(b"protect within commercially", response.data)
        self.assertIn(
            b"to prevent loss and theft, as well as unauthorized", response.data
        )
        self.assertIn(
            b"access, disclosure, copying, use or modification.", response.data
        )
        self.assertIn(
            b"Your continued use of our website will be regarded as acceptance",
            response.data,
        )
        self.assertIn(
            b"practices around privacy and personal information", response.data
        )
        self.assertIn(b"If you have any questions about", response.data)
        self.assertIn(
            b"how we handle user data and personal information", response.data
        )

File: browser_test/test_Privacy_Policy.js
web.init();
web.open('http://127.0.0.1:5000/');
web.selectWindow('title=Set Now');
//               id: id=policy_page_link
//       xpath:link: //a[contains(text(),\'Privacy Policy\')]
// xpath:attributes: //a[@id=\'policy_page_link\']
// xpath:idRelative: //div[@id=\'footercontent\']/span[14]/a
//       xpath:href: //a[@href=\'http://127.0.0.1:5000/privacy-policy\']
//   xpath:position: //span[14]/a
//             link: link=Privacy Policy
//              css: css=#policy_page_link
web.click('id=policy_page_link');
web.selectWindow('title=Privacy Policy');
//               id: id=policy_content
// xpath:attributes: //div[@id=\'policy_content\']
// xpath:idRelative: //div[@id=\'wrapper\']/div
//   xpath:position: //div[2]/div
//              css: css=#policy_content
web.click('id=policy_content');
// xpath:idRelative: //div[@id=\'policy_content\']/p[3]
//   xpath:position: //p[3]
//              css: css=#policy_content > p:nth-child(3)
web.assertText('//div[@id=\'policy_content\']/p[3]', `We only retain collected information for as long as necessary to provide you with your requested service.
 What data we store, we’ll protect within commercially acceptable means to prevent loss and theft, as
 well as unauthorized access, disclosure, copying, use or modification.`);
// xpath:idRelative: //div[@id=\'page_title\']/div/h2
//   xpath:position: //div/div[1]/div/h2
//              css: css=#page_title > div > h2
web.click('//div[@id=\'page_title\']/div/h2');
// xpath:idRelative: //div[@id=\'page_title\']/div/h2
//   xpath:position: //div/div[1]/div/h2
//              css: css=#page_title > div > h2
web.assertText('//div[@id=\'page_title\']/div/h2', 'Privacy Policy');
//   xpath:position: //body/div
//              css: css=body > div
web.click('//body/div');
web.assertTitle('Privacy Policy');
//   xpath:position: //body/div
//              css: css=body > div
web.click('//body/div');
web.open('http://127.0.0.1:5000/');
